apps/web/views.py

```diff
@@ -35,29 +35,35 @@ def internal_server_error(request):
 
 
 @throttle_classes([AnonRateThrottle, ])
-@api_view(['POST', ])
+@api_view(['GET', 'POST'])
 @permission_classes((permissions.AllowAny,))
 def contact_us(request):
     try:
         user = User.objects.get(username=request.user)
         name = user.username
         email = user.email
-        request_data = {"name": name, "email": email}
-        request_data['message'] = request.data['message']
-        serializer = ContactSerializer(data=request_data)
+    except:
+        serializer = ContactSerializer(data=request.data)
```

Comment:
I think something like this can. Let me know your concerns if any

```
user_does_not_exist = False
try:
    user = User.objects.get(username=request.user)
    name = user.username
    email = user.email
    request_data = {'name': name, 'email': email}
except:
    request_data = request.data
    user_does_not_exist = True

if request.method == 'POST' or user_does_not_exist:
    if request.POST.get('message'):
        request_data['message'] = request.POST.get('message')
    serializer = ContactSerializer(data=request_data)
    if serializer.is_valid():
        serializer.save()
        response_data = {'message': 'We have received your request and will contact you shortly.'}
        return Response(response_data, status=status.HTTP_201_CREATED)
    return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)

elif request.method == 'GET':
    response_data = {"name": name, "email": email}
    return Response(response_data, status=status.HTTP_200_OK)
```